File: mefgl/problems_and_solutions/Task1_nucl_count/Antonia_count_nucleotides.py

#open file and read text, print contents
my_file = open("rosalind_dna.txt")
dna_string = my_file.read()
print(dna_string)

#count characters
A_count = dna_string.count('A')
C_count = dna_string.count('C')
G_count = dna_string.count('G')
T_count = dna_string.count('T')

#print counts
print(str(A_count) + " " + str(C_count) + " " + str(G_count) + " " + str(T_count))

#print counts with headers
print("A: " + str(A_count))
print("C: " + str(C_count))
print("G: " + str(G_count))
print("T: " + str(T_count))

File: mefgl/problems_and_solutions/Task1_nucl_count/Antonia_count_nucleotides.R
## housekeeping
rm(list=ls())

#load packages
library(stringr)

#read in file and print text
dna_string=scan("rosalind_dna.txt", what="character")
dna_string

#counts
A_count = str_count(dna_string, "A")
C_count = str_count(dna_string, "C")
G_count = str_count(dna_string, "G")
T_count = str_count(dna_string, "T")

#return counts with spaces
dna_counts=cat(A_count, C_count, G_count, T_count)